python/src/equistore/operations/pow.py

```diff
@@ -45,16 +43,25 @@ def _pow_block_constant(block: TensorBlock, constant: float) -> TensorBlock:
         components=block.components,
         properties=block.properties,
     )
+    if len(block.gradients_list()) == 0:
+        return result_block
+
+    _shape = ()
+    for c in block.components:
+        _shape += (len(c),)
+    _shape += (len(block.properties),)
 
     for parameter, gradient in block.gradients():
         values_grad = []
-        for isample in range(len(block.samples)):
-            isample_grad1 = np.where(gradient.samples["sample"] == isample)[0]
-            values_grad.append(
-                constant
-                * gradient.data[isample_grad1]
-                * block.values[isample] ** (constant - 1)
+        diff_components = len(gradient.components) - len(block.components)
```

Comment:
A tiny optimization (not sure it will have any effects though)

```suggestion
        gradient_data = gradient.data
        diff_components = len(gradient_data.shape) - len(block.values.shape)
```

Reply:
gradient_data.shape is not the len of the components

Reply:
same for len(block.values.shape) != len(block.components)

maybe we need to put
`len(gradient_data.shape) - len(block.values.shape)-2`

Reply:
sorry you are right 
`diff_components = len(gradient_data.shape) - len(block.values.shape)` works